common/arch/cocoa/SDLMain.m

```diff
@@ -351,12 +351,43 @@ - (NSString *)stringByReplacingRange:(NSRange)aRange with:(NSString *)aString
 #endif
 
 
+static int IsRootCwd()
+{
+    char buf[MAXPATHLEN];
+    char *cwd = getcwd(buf, sizeof (buf));
+    return (cwd && (strcmp(cwd, "/") == 0));
+}
+
+static int IsTenPointNineOrLater()
+{
+    /* Gestalt() is deprecated in 10.8, but I don't care. Stop using SDL 1.2. */
+    SInt32 major, minor;
+    Gestalt(gestaltSystemVersionMajor, &major);
+    Gestalt(gestaltSystemVersionMinor, &minor);
+    return ( ((major << 16) | minor) >= ((10 << 16) | 9) );
```

Comment:
This seems like a strange way to test the version.  It'd probably generate simpler code, and definitely be more readable, to test the version components individually.

Reply:
I think the only way of doing that would be to use a more recent function that tells the OS version. Unfortunately the methods listed on [a StackExchange article](https://stackoverflow.com/questions/11072804/how-do-i-determine-the-os-version-at-runtime-in-os-x-or-ios-without-using-gesta) look to be even more complicated.

Currently DXX-Rebirth is supported on Mac OS X 10.6 and higher - so
`[[NSProcessInfo processInfo] operatingSystemVersion]`
is out.

Reply:
This code was ripped straight out of SDL 1.2's SDLMain library.

Reply:
If this is a backport of upstream's fix, that is an excellent reason to leave it as-is.  I'm inclined to track their code absent a compelling reason to deviate (e.g. if it caused a regression on a supported platform - which does not apply here).